go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +151,110 @@ func coverageShowMetricsUpdater(component string) {
 	}
 }
 
+// registerStopwatchShowMetrics registers stopwatch/show metrics for
+// various components(ovn-northd, ovn-controller, and ovs-vswitchd) with prometheus
+func registerStopwatchShowMetrics(component string, metricNamespace string, metricSubsystem string) {
+	stopwatchShowMetricsMap := componentStopwatchShowMetricsMap[component]
+	for metricName, metricInfo := range stopwatchShowMetricsMap {
+		metricInfo.metric = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      metricName,
+			Help:      metricInfo.help,
+		})
+		prometheus.MustRegister(metricInfo.metric)
+	}
+}
+
+// getStopwatchShowOutputMap obtains the stopwatch/show metric values for the specified component.
+func getStopwatchShowOutputMap(component string) (map[string]int, error) {
+	var stdout, stderr string
+	var err error
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("recovering from a panic while parsing the stopwatch/show output for %s: %v", component, r)
+		}
+	}()
+
+	switch component {
+	case ovnController:
+		stdout, stderr, err = util.RunOVNControllerAppCtl("stopwatch/show")
```

Comment:
If we add/remove/change a metric in ovn, how will that affect this?

Reply:
If we want to have the new metric provided by ovn-k too, we would have to add it in https://github.com/ovn-org/ovn-kubernetes/blob/8406f70c24b3936dd301a9a781e856a3f0cf8963/go-controller/pkg/metrics/ovn.go#L213-L241 (or https://github.com/ovn-org/ovn-kubernetes/blob/8406f70c24b3936dd301a9a781e856a3f0cf8963/go-controller/pkg/metrics/ovn_northd.go#L76-L81) where we can also provide a "help/description" text for the metric.

The thing is, we need to register the metrics to Prometheus at start up. So we could run a `ovn-appctl stopwatch/show` at the start up to get all the metric names, but I wanted to orient at the implementation how it was done for the metrics collection of `coverage/show`.
But of course, I can change it.